### PR TITLE
Implemented Notify functionality

### PR DESCRIPTION
**Pull request by woutermeuwis**

### CODE REVIEW

**Review by SelleslaghGianni — APPROVED** [review]
I'd add a small message telling the user the action is canceled when he chose to cancel the action (when not subbed) but that's really not that important imo :)

**Comment by woutermeuwis** [comment]
> I'd add a small message telling the user the action is canceled when he chose to cancel the action (when not subbed) but that's really not that important imo :)

Would be a nice addition though, will implement this 👍

**Review by SelleslaghGianni — APPROVED** [review]
:)

File: commands/notify.js [hunk 1]
```diff
@@ -137,10 +138,159 @@ async function DisplayListSelector (interaction, guild, question) {
...
+    if (interaction.customId === 'cancel') {
+      await reply(interaction, 'Ale, dag en bedankt dan eh!')
+      return
+    } else if (interaction.customId === 'sub') {
+      guild.notifyLists[listName].push(userId)
+      await reply(interaction, `Je ontvangt nu notificaties voor de lijst ${listName}`)
+      return
+    } else {
+      // continue and notify for list
+    }
```
Comment (FreTimmerman) [inline commands/notify.js:246]: is something missing here?  if no, then :
```suggestion
    } 
```

File: commands/notify.js [hunk 2]
```diff
@@ -137,10 +138,159 @@ async function DisplayListSelector (interaction, guild, question) {
...
+
+  // notify list
+  const botName = await getUserNameFromIdAsync(interaction, interaction.applicationId)
+  const username = await getTagFromIdAsync(interaction, userId)
+  const tags = []
+  let tag = ''
+  for (const id of subscribers) {
+    const usr = await getTagFromIdAsync(interaction, id)
+
+    if (usr.length + tag.length + 5 < DISCORD_MSG_MAX_LENGTH) {
```
Comment (FreTimmerman) [inline commands/notify.js:262]: where does this magic `5` come from?

**Review by FreTimmerman — DISMISSED** [review]

File: commands/notify.js [hunk 3]
```diff
@@ -137,10 +138,159 @@ async function DisplayListSelector (interaction, guild, question) {
...
+  }
+
+  if (subscribers.length === 0) {
+    await reply(interaction, `Er is nog niemand ingeschreven op deze lijst, ${foemp()}!`)
+    return
+  }
+
+  // notify list
+  const botName = await getUserNameFromIdAsync(interaction, interaction.applicationId)
+  const username = getTagFromId(interaction, userId)
```
Comment (FreTimmerman) [inline commands/notify.js:255]: watch out. you removed the `interaction` parameter from the function, only `userId` needed here

File: commands/notify.js [hunk 4]
```diff
@@ -137,10 +138,159 @@ async function DisplayListSelector (interaction, guild, question) {
...
+  const tags = []
+  let tag = ''
+  const connection = ', '
+  for (const id of subscribers) {
+    const usr = getTagFromId(interaction, id)
```
Comment (FreTimmerman) [inline commands/notify.js:260]: watch out. you removed the `interaction` parameter from the function, only `userId` needed here

**Review by FreTimmerman — APPROVED** [review]

**Review by FreTimmerman — APPROVED** [review]

**Pull request merged**